neuralop/layers/spectral_convolution.py

```diff
@@ -416,10 +416,14 @@ def forward(
 
         if self.complex_data:
             x = torch.fft.fftn(x, norm=self.fft_norm, dim=fft_dims)
+            fft_shift_dims = fft_dims
         else: 
             x = torch.fft.rfftn(x, norm=self.fft_norm, dim=fft_dims)
+            fft_shift_dims = fft_dims[:-1] # for simplicity, don't shift the last dimension when x is real in spatial domain
```

Comment:
It's not for simplicity - we use rfft so the redundant part is simply not there in the last dim.